cmd/policy.go

```diff
@@ -83,12 +84,27 @@ func (sys *PolicySys) Init(objAPI ObjectLayer) error {
 	for _, bucket := range buckets {
 		config, err := GetPolicyConfig(objAPI, bucket.Name)
 		if err != nil {
-			if _, ok := err.(BucketPolicyNotFound); !ok {
-				return err
+			if _, ok := err.(BucketPolicyNotFound); ok {
+				continue
 			}
-		} else {
-			sys.Set(bucket.Name, *config)
+			return err
+		}
+
+		// This part is specifically written to handle migration
+		// when the Version string is empty, this was allowed
+		// in all previous minio releases but we need to migrate
+		// those policies by properly setting the Version string
+		// from now on.
+		if config.Version == "" {
+			logger.Info("Found in-consistent bucket policies, Migrating them for Bucket: (%s)", bucket.Name)
+			config.Version = policy.DefaultVersion
+		}
+
+		if err = savePolicyConfig(objAPI, bucket.Name, config); err != nil {
```

Comment:
You have to save the policy only if `config.Version == ""` case. otherwise its unnecessarily.